### DISCUSSION TITLE
ORAN DU/CU Integration Feature

### DISCUSSION THREAD

**Discussion by infinitydon** [HewlettPackard/PacketRusher#67] [Ideas]
@linouxis9 - Reading the latest update from srsran having a high performance DU/CU software (https://docs.srsran.com/projects/project/en/latest/tutorials/source/dpdk/source/index.html).

Will this be an interesting feature to add to packet-rusher by adding support for ORAN RU and UE scenario? Essentially giving 2 options for the RAN:
- gNB and UE
- RU and UE

Thanks

**Comment by linouxis9** [comment]
Hi Christopher @infinitydon!

Thanks for sharing the latest update from srsRAN.
However, I feel adding support for things like ORAN, DU/CU split is counter productive for PacketRusher.

The main goals of PacketRusher is to be able to easily test a 5G Core externally as if it were a blackbox. For a 5GC, call flows with the radio do not change much if you have an all-in-one gNodeB or something ORAN-based, or with a DU/CU split: in any cases traffic will flow through NGAP and GTP-U from the radio to the core.
The goal of PacketRusher is not necessarily to implement gNodeB features which do not make a difference for the 5GC, it's better if we focus making features that test differents aspects of the 5GC.
Even worse, these kind of features could impact negatively the performance/footprint of PacketRusher while performance is one of its goals.

What do you think?

Thanks!
Valentin

**Comment by infinitydon** [comment]
@linouxis9  - If the focus of packetRusher is strictly to test the 5G core components then I agree though an interesting perspective will be to able to stress-test the GTP-U from the CU-UP all the way to the UPF.
